### ISSUE TITLE
When the RunTemplate has been modified, Pipeline Service should reconcile and stamp out a new object

### ISSUE THREAD

**Issue by emmjohnson** [vmware-tanzu/cartographer#83]
Similar to: https://github.com/vmware-tanzu/cartographer/blob/8b29f73c62132ccf28c53ad035a3cfb24463a797/pkg/registrar/registrar.go#L90

Origin: https://github.com/vmware-tanzu/cartographer/issues/6
-----

```
as an operator
when I modify a RunTemplate
then all pipelines that refer to that runtemplate, stamp out new objects
```

Question:
As a developer, how do I know the difference between two pipeline runs for the same sha?

```gherkin
given I have a pipeline built against a template
when I modify the template
then I see another build
```

**Comment by DanieldeR** [comment]
Before IPM this story should have some acceptance criteria. @emmjohnson can you add some acceptance criteria to this issue?

**Comment by DanieldeR** [comment]
As discussed at IPM, for the time being, in the short term, we'll use observed generation to differentiate the case where there are multiple pipeline runs that are stamped out.

If a runTemplate references a pipelineRef, we won't be handling this case. We'll need a separate issue to handle this use case.

**Comment by emmjohnson** [comment]
> As discussed at IPM, for the time being, in the short term, we'll use observed generation to differentiate the case where there are multiple pipeline runs that are stamped out.
> 
> If a runTemplate references a pipelineRef, we won't be handling this case. We'll need a separate issue to handle this use case.

PipelineRef here refers to a TektonPipeline

**Comment by waciumawanjohi** [comment]
Closed by #109
